docs/sources/helm-charts/mimir-distributed/configure/configure-autoscaling.md

```diff
@@ -0,0 +1,194 @@
+
+1. If the Mimir/GEM cluster becomes unavailable, autoscaling will stop working because it cannot query the metrics
+2. This can prevent the cluster from automatically scaling up during high load or recovery
+3. The inability to scale can further exacerbate the cluster's unavailability
+4. This may prevent the cluster from recovering
+
+Instead, use a separate Prometheus instance or a different metrics backend for autoscaling metrics.
+{{< /admonition >}}
+
+## Supported components
+
+Mimir Helm chart supports autoscaling for the following components:
+
+- [Querier](https://grafana.com/docs/mimir/<MIMIR_VERSION>/references/architecture/components/querier/)
+- [Query-frontend](https://grafana.com/docs/mimir/<MIMIR_VERSION>/references/architecture/components/query-frontend/)
+- [Distributor](https://grafana.com/docs/mimir/<MIMIR_VERSION>/references/architecture/components/distributor/)
+
+## About KEDA
+
+KEDA is a Kubernetes operator that simplifies the setup of HPA with custom metrics from Prometheus. It consists of:
+
+- An operator and external metrics server
+- Support for multiple metric sources, including Prometheus
+- Custom resources (`ScaledObject`) that define scaling parameters
+- Automatic HPA resource management
+
+For more information, refer to [KEDA documentation](https://keda.sh).
+
+## Configure autoscaling for a new installation
+
+This section describes how to enable autoscaling when deploying Mimir for the first time.
+
+### Steps
+
+1. Configure the Prometheus metrics source in your values file:
+
+   ```yaml
+   kedaAutoscaling:
+     prometheusAddress: "http://prometheus.monitoring:9090"
+     pollingInterval: 10
+   ```
+
+2. Enable and configure autoscaling for desired components:
+
+   ```yaml
+   querier:
+     kedaAutoscaling:
+       enabled: true
+       minReplicaCount: 2
+       maxReplicaCount: 10
+   ```
+
+3. Deploy Mimir using Helm:
+   ```bash
+   helm upgrade --install mimir grafana/mimir-distributed -f values.yaml
+   ```
+
+### Expected outcome
+
+After deployment:
+
+- KEDA creates ScaledObject resources for configured components
+- HPA resources are automatically created and begin monitoring metrics
+- Components scale based on configured thresholds and behaviors
+
+## Migrate existing deployments to autoscaling
+
+This section describes how to safely enable autoscaling for an existing Mimir deployment.
+
+{{< admonition type="warning" >}}
+Migrating to autoscaling carries risks for cluster availability:
+
+1. Autoscaling support in the Helm chart is currently experimental
+2. Enabling autoscaling removes the `replicas` field from Deployments
+3. If KEDA/HPA hasn't started autoscaling the Deployment yet, Kubernetes interprets no replicas as meaning 1 replica
+4. This can cause an outage if the transition is not handled carefully
+5. If you're using GitOps tools like FluxCD or ArgoCD, additional steps might be required to handle the transition smoothly
```

Comment:
```suggestion
Autoscaling support in the Helm chart is currently experimental. Migrating to autoscaling carries risks for cluster availability.

Enabling autoscaling removes the `replicas` field from deployments. If KEDA/HPA hasn't started autoscaling a deployment yet, Kubernetes interprets no replicas as meaning 1 replica. This can cause an outage if the transition is not handled carefully. If you're using GitOps tools like FluxCD or ArgoCD, you might need to take additional steps to manage the transition.
```

Reply:
Same comment as above re. a list vs. paragraph.